pkg/go/transformer/dsltojson.go

```diff
@@ -66,37 +71,47 @@ func (l *openFgaDslListener) ExitTypeDef(ctx *parser.TypeDefContext) {
 	l.currentTypeDef = nil
 }
 
-func (l *openFgaDslListener) EnterRelationDeclaration(_ctx *parser.RelationDeclarationContext) {
+func (l *OpenFgaDslListener) EnterRelationDeclaration(_ctx *parser.RelationDeclarationContext) {
 	l.currentRelation = &relation{
 		Rewrites: []*pb.Userset{},
 		TypeInfo: pb.RelationTypeInfo{DirectlyRelatedUserTypes: []*pb.RelationReference{}},
 	}
 }
 
-func (l *openFgaDslListener) ExitRelationDeclaration(ctx *parser.RelationDeclarationContext) {
+func (l *OpenFgaDslListener) ExitRelationDeclaration(ctx *parser.RelationDeclarationContext) {
+	if ctx.RelationName() == nil {
+		return
+	}
+
 	relationName := ctx.RelationName().GetText()
 
+	if l.currentRelation.Rewrites == nil || len(l.currentRelation.Rewrites) == 0 {
```

Comment:
Do we need to check for whether l.currentRelation is null?

Reply:
Already initialized here: https://github.com/openfga/language/blob/50b003d312da3fdb52812a562ea22292bc186e6e/pkg/go/transformer/dsltojson.go#L74